sdk/ai/azopenai/models.go

```diff
@@ -618,68 +787,56 @@ type CompletionsUsage struct {
-	Sexual *ContentFilterResultsSexual
+	Sexual *ContentFilterResult
 
 	// Describes language related to physical actions intended to hurt, injure, damage, or kill someone or something; describes
 	// weapons, etc.
-	Violence *ContentFilterResultsViolence
-}
-
-// ContentFilterResultsHate - Describes language attacks or uses that include pejorative or discriminatory language with reference
-// to a person or identity group on the basis of certain differentiating attributes of these groups
-// including but not limited to race, ethnicity, nationality, gender identity and expression, sexual orientation, religion,
-// immigration status, ability status, personal appearance, and body size.
-type ContentFilterResultsHate struct {
-	// REQUIRED; A value indicating whether or not the content has been filtered.
-	Filtered *bool
-
-	// REQUIRED; Ratings for the intensity and risk level of filtered content.
-	Severity *ContentFilterSeverity
+	Violence *ContentFilterResult
 }
 
-// ContentFilterResultsSelfHarm - Describes language related to physical actions intended to purposely hurt, injure, or damage
-// one’s body, or kill oneself.
-type ContentFilterResultsSelfHarm struct {
-	// REQUIRED; A value indicating whether or not the content has been filtered.
-	Filtered *bool
-
-	// REQUIRED; Ratings for the intensity and risk level of filtered content.
-	Severity *ContentFilterSeverity
-}
+// ContentFilterResultsError - Describes an error returned if the content filtering system is down or otherwise unable to
+// complete the operation in time.
+type ContentFilterResultsError struct {
+	// REQUIRED; One of a server-defined set of error codes.
+	Code *string
 
-// ContentFilterResultsSexual - Describes language related to anatomical organs and genitals, romantic relationships, acts
-// portrayed in erotic or affectionate terms, physical sexual acts, including those portrayed as an assault or a
-// forced sexual violent act against one’s will, prostitution, pornography, and abuse.
-type ContentFilterResultsSexual struct {
-	// REQUIRED; A value indicating whether or not the content has been filtered.
-	Filtered *bool
+	// REQUIRED; A human-readable representation of the error.
+	Message *string
 
-	// REQUIRED; Ratings for the intensity and risk level of filtered content.
-	Severity *ContentFilterSeverity
-}
+	// An array of details about specific errors that led to this reported error.
+	Details []azureCoreFoundationsError
 
-// ContentFilterResultsViolence - Describes language related to physical actions intended to hurt, injure, damage, or kill
-// someone or something; describes weapons, etc.
-type ContentFilterResultsViolence struct {
-	// REQUIRED; A value indicating whether or not the content has been filtered.
-	Filtered *bool
+	// An object containing more specific information than the current object about the error.
+	Innererror *azureCoreFoundationsErrorInnererror
```

Comment:
This looks suspicious.

Reply:
Fixed. That type sneaks in sometimes.

I added some autorest hackery and it looks better now. Also, fixed the InnerErrorInnerError nonsense (the type is actually recursive).